src/nway/onprem_utils/query_utils.py

```diff
@@ -0,0 +1,104 @@
+import os
+import pg8000
+
+
+class CredentialsException(Exception):
+    pass
+
+
+def get_db_credentials(
+        env_prefix="LIMS_", host='limsdb2',
+        database='lims2', port=5432) -> dict:
+    """Get DB credentials from environment variables and keyword
+    args. Environment variables must be in the format `{env_prefix}{key}`,
+    where `key` is one of the following (case-sensitive):
+    ['USER', 'PASSWORD', 'HOST', 'PORT', 'DATABASE'].
+    Examples: LIMS_USER, LIMS_PASSWORD.
+    Values for `user` and `password` must be environment variables.
+    Values for `host`, `port`, and `database` can be either environment
+    variables or keyword arguments, with environment variables taking
+    precedence.
+
+    Parameters
+    ----------
+    env_prefix : str
+        expected environment variable prefix for credential keys.
+    host : str
+        default host value, if not found in os.environ
+    database : str
+        default database value, if not found in os.environ
+    port : int
+        default port value, if not found in os.environ
+
+    Returns
+    -------
+    dict
+        A dictionary of DB credentials. Contains the following fields:
+        'user', 'host', 'database', 'password', 'port'.
+
+    Raises
+    ------
+    CredentialsException
+        Raised if user and password are not set as ENV variables
+    """
+
+    db_credentials = {}
+    for key in ['user', 'password']:
+        env_key = env_prefix + key.upper()
+        try:
+            db_credentials[key] = os.environ[env_key]
+        except KeyError:
+            raise CredentialsException(f"ENV variable {env_key} must be set")
+
+    defaults = {'host': host, 'port': port, 'database': database}
+
+    for key in ['host', 'port', 'database']:
+        env_key = env_prefix + key.upper()
+        db_credentials[key] = os.environ.get(env_key, defaults[key])
+        if db_credentials[key] is None:
+            raise CredentialsException(
+                    "no ENV variable found and no default provided "
+                    f"for {env_key}")
+
+    return db_credentials
+
+
+class DbConnection():
```

Comment:
I feel like I've seen this code repeated quite often, we're a bit short on time currently, but maybe create a ticket that documents all the repeated instances so we can consolidate them into a utility package/repository...

Reply:
yeah, it is all over the place, including a stand-alone: https://github.com/AllenInstitute/simple-lims-connection
but, that repo is not public. I suppose this `[ONPREM]` option could require that private install, and only local users could do it anyway,